### PR TITLE
fix(types): external module type error (CompoundedComponent)

### PR DESCRIPTION
**Pull request by wangcch**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/33929
https://github.com/ant-design/ant-design/issues/33091

### 💡 Background and solution

Wrong type for `CompoundedComponent` from external module
`type` vs `interface` alias

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix 'CompoundedComponent' type error from external module   |
| 🇨🇳 Chinese |  修复外部暴露类 `CompoundedComponent` 的组建的类型报错  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/badge/index.tsx [hunk 1]
```diff
@@ -133,7 +133,7 @@ const Badge: CompoundedComponent = ({
   const displayNode =
     !livingCount || typeof livingCount !== 'object'
       ? undefined
-      : cloneElement(livingCount, oriProps => ({
+      : cloneElement(livingCount, (oriProps) => ({
```
Comment (wangcch) [inline components/badge/index.tsx:136]: Hi 👋
And I found that `.prettierrc` (`"arrowParens": "avoid"`) and rome js config seem to be in conflict.

pre-commit will execute rome-format before

**Comment by afc163** [comment]
Conflicting

**Comment by wangcch** [comment]
> Conflicting

So I don't need to rebase the commit ...👇
https://github.com/ant-design/ant-design/pull/38666/commits/11775522856e1fb4fd51dee008763fffe7670fae

**Comment by li-jia-nan** [comment]
rebase

**Comment by wangcch** [comment]
> rebase

ok. and thanks~🙆‍♂️

**Comment by hengkx** [comment]
rebase

**Comment by wangcch** [comment]
> rebase

ok 👌

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**